File: problems/1952-three-divisors/solution.py
class Solution:
    def isThree(self, n: int) -> bool:
        if n <= 3:
            return False
        
        count = 0
        for i in range(1, int(n**0.5) + 1):
            if n % i == 0:
                if n // i == i:
                    count += 1
                else:
                    count += 2
        
        return count == 3

File: problems/1952-three-divisors/solution.c
#include <stdbool.h>
#include <math.h>

bool isThree(int n) {
    if (n <= 3) {
        return false;
    }
    int count = 0;
    for (int i = 1; i * i <= n; i++) {
        if (n % i == 0) {
            if (i * i == n) {
                count++;
            } else {
                count += 2;
            }
        }
    }
    return count == 3;
}